plugins/device/cmdexec/device.go

```diff
@@ -0,0 +1,51 @@
+package cmdexec
+
+import (
+	"os/exec"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func createVgName(device string) string {
+	vgName := strings.Replace("vg"+device, "/", "-", -1)
+	return vgName
+}
+
+// DeviceSetup is used to prepare device before using devices.
+func DeviceSetup(device string) error {
+
+	var err error
+	defer func() {
+		if err != nil {
+			DeviceCleanup(device)
+		}
+	}()
+	pvcreateCmd := exec.Command("pvcreate", "--metadatasize=128M", "--dataalignment=256K", device)
+	if err := pvcreateCmd.Run(); err != nil {
+		log.WithError(err).WithField("device", device).Error("pvcreate failed for device")
```

Comment:
These loggers must have context about request and/or txn. You can pass the logger to this function.